src/micro_component.rs

```diff
@@ -0,0 +1,234 @@
+use gtk::glib::{self, Sender};
+
+use std::{
+    cell::{BorrowError, BorrowMutError, Ref, RefCell, RefMut}, 
+    fmt::{Debug, Display}, 
+    rc::Rc
+};
+
+/// Errors which might get returned from [`MicroComponent::update_view`] method
+/// 
+/// Both [`std::fmt::Display`] and [`std::fmt::Debug`] are implemented for this
+/// enum so if you just would like to show an error you don't need to do the
+/// matching.
+pub enum MicroComponentError {
+    /// Error returned if borrow failed
+    Borrow(BorrowError),
+    /// Error returned if borrowing mutably failed
+    BorrowMut(BorrowMutError),
+}
+
+/// Helper to convert values of [`std::cell::BorrowError`] into [`MicroComponentError`]
+impl From<BorrowError> for MicroComponentError {
+    fn from(err: BorrowError) -> Self {
+        MicroComponentError::Borrow(err)
+    }
+}
+
+/// Helper to convert values of [`std::cell::BorrowMutError`] into [`MicroComponentError`]
+impl From<BorrowMutError> for MicroComponentError {
+    fn from(err: BorrowMutError) -> Self {
+        MicroComponentError::BorrowMut(err)
+    }
+}
+
+/// Formats [`MicroComponentError`] for empty format `{}`
+/// 
+/// This allows you to print errors without doing `matching` or `if let` statements
+impl Display for MicroComponentError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            MicroComponentError::Borrow(err) => {
+                f.write_fmt(format_args!("{}", err))
+            },
+            MicroComponentError::BorrowMut(err) => {
+                f.write_fmt(format_args!("{}", err))
+            }
+        }
+    }
+}
+
+/// Formats [`MicroComponentError`] for debug format `{:?}`
+/// 
+/// This allows you to print errors without doing `matching` or `if let` statements
+impl Debug for MicroComponentError {
```

Comment:
Is there a specific reason to manually implement this? I think using `#derive[Debug]` should do the same.